InterfacePython/testAddons.py

```diff
@@ -136,8 +136,43 @@ def checkUnstructuredL2(self):
 
         self.assertLess(np.max(np.abs(model - surrogate[:,0])), 5.E-3, "Constructed from unstructured too inaccurate.")
 
+    def checkExoticQuadrature(self):
+        # Initialize some useful constants and utility functions.
+        integrand = lambda x : np.exp(-np.linalg.norm(x) ** 2)
+        sinc1s0 = lambda x : np.sinc(x / np.pi)
+        sinc10s1 = lambda x : np.sinc(10 * (x - 1) / np.pi)
+        max_level = 25
+        nref = 101
+        def create_surrogate(lambda_fn):
+            grid = Tasmanian.makeGlobalGrid(1, 1, nref, "level", "gauss-legendre")
+            grid.loadNeededValues(np.array([lambda_fn(x) for x in grid.getNeededPoints()]))
+            return grid
+        def compute_integral(grid, integrand):
+            return np.sum([integrand(x) for x in grid.getPoints()] * grid.getQuadratureWeights())
```

Comment:
As a general rule in Python, it is better to avoid entry-wise operations and use full arrays. For example, all the integrand functions accept vectors so you can avoid the for-loop:
```
return np.sum(integrand(grid.getPoints().squeeze()) * grid.getQuadratureWeights())
```

Reply:
I modified this to
```
return np.sum(np.apply_along_axis(integrand, 1, grid.getPoints()) * grid.getQuadratureWeights())
```
because the `integrand` function has different behaviors when applied pointwise than to a vector of points (generated by `np.squeeze()`.